ksql-version-metrics-client/src/main/java/io/confluent/ksql/version/metrics/KsqlVersionCheckerAgent.java

```diff
@@ -19,48 +19,70 @@
 import io.confluent.ksql.version.metrics.collector.KsqlModuleType;
 import io.confluent.support.metrics.BaseSupportConfig;
 import io.confluent.support.metrics.PhoneHomeConfig;
+import java.time.Clock;
+import java.util.Objects;
+import java.util.Optional;
 import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class KsqlVersionCheckerAgent implements VersionCheckerAgent {
 
-  private KsqlVersionChecker ksqlVersionChecker;
+  private static final long MAX_INTERVAL = TimeUnit.DAYS.toMillis(1);
+
 
   private boolean enableSettlingTime;
 
+  private Clock clock;
+
+  private volatile long requestTime;
+  private final Supplier<Boolean> activeQuerySupplier;
+
   private static final Logger log = LoggerFactory.getLogger(KsqlVersionCheckerAgent.class);
 
-  public KsqlVersionCheckerAgent() {
-    this(true);
+  public KsqlVersionCheckerAgent(final Supplier<Boolean> activeQuerySupplier) {
+    this(activeQuerySupplier, true, Clock.systemDefaultZone());
   }
 
-  KsqlVersionCheckerAgent(final boolean enableSettlingTime) {
+  KsqlVersionCheckerAgent(
+      final Supplier<Boolean> activeQuerySupplier,
+      final boolean enableSettlingTime,
+      final Clock clock) {
     this.enableSettlingTime = enableSettlingTime;
+    this.activeQuerySupplier = Objects.requireNonNull(activeQuerySupplier, "activeQuerySupplier");
+    this.clock = clock;
   }
 
   @Override
   public void start(final KsqlModuleType moduleType, final Properties ksqlProperties) {
+    start(null, moduleType, ksqlProperties);
+  }
+
+  void start(
```

Comment:
Another rule for you list: Exposing methods like this, just for tests to call, is not ideal.  When you want to do something like this... stop... and think how you can use _dependency injection_ to solve this testing conundrum.  

I recommend that you read up on Google about dependency injection.  Not the dependency injection frameworks like Spring / Guice, but just straight dependency injection, especially in the context of improving the test-ability of your code.

In this situation, inject a factory for creating the `KsqlVersionChecker` instance into this classes constructor.

This allows you to:
 - verify the factory, (and hence the `KsqlVersionChecker` constructor), is being passed the correct params
 - inject a mock `KsqlVersionChecker` into the class, so that you can control what it does.


```java
  public KsqlVersionCheckerAgent(final Supplier<Boolean> activeQuerySupplier) {
    this(activeQuerySupplier, true, Clock.systemDefaultZone(), KsqlVersionChecker::new);
  }

KsqlVersionCheckerAgent(
      final Supplier<Boolean> activeQuerySupplier,
      final boolean enableSettlingTime,
      final Clock clock,
      final VerisonCheckerFactory versionCheckerFactory
) {
  ...
}

@Override
  public void start(final KsqlModuleType moduleType, final Properties ksqlProperties) {
    final BaseSupportConfig ksqlVersionCheckerConfig =
        new PhoneHomeConfig(ksqlProperties, "ksql");

    if (!ksqlVersionCheckerConfig.isProactiveSupportEnabled()) {
      log.warn(legalDisclaimerProactiveSupportDisabled());
      return;
    }
    
    final KsqlVersionChecker ksqlVersionChecker = versionCheckerFactory.create(
            ksqlVersionCheckerConfig,
            moduleType,
            enableSettlingTime,
            this::isActive);

   ...
}

@FunctionalInterface
interface VerisonCheckerFactory {
   KsqlVersionChecker create(
      // String threadName,  <-- don't add this one, it's constant! Just remove it from the `KsqlVersionChecker` class
      // boolean isDaemon, <-- don't add this one, it's always true. Just remove this from the `KsqlVersionChecker` constructor
      BaseSupportConfig ksqlVersionCheckerConfig,
      //    Runtime serverRuntime, <-- don't add this one. Just use `Runtime.getRuntime()` directly inside the `KsqlVersionChecker` class
      KsqlModuleType moduleType,
      boolean enableSettlingTime,
      Supplier<Boolean> activenessStatusSupplier
   );
}
```